### PR TITLE
[fix](mtmv) Fix generate hyper graph wrongly when has filter which can not push down

### PR DESCRIPTION
**Pull request by seawinde**
### What problem does this PR solve?

When query is as following:
```sql
 select
              o_orderdate,
              o_shippriority,
              o_comment,
              l_orderkey, 
              l_partkey
            from
              orders left
              join lineitem on l_orderkey = o_orderkey
              left join partsupp on ps_partkey = l_partkey and l_suppkey = ps_suppkey;
```
hyper graph is as following:
this` leftExtendedNodes` Shoule be contained both orders and lineitem, which should be 3, but now is 2
image

Issue Number: close #xxx

Related PR: https://github.com/apache/doris/pull/28006

Problem Summary:

This may cause associate wrongly when use hyper graph and materialized view which depends on it rewrite wrongly

### Release note
Fix generate hyper graph wrongly when has filter which can not push down

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [x] Confirm the release note
- [x] Confirm test cases
- [x] Confirm document
- [x] Add branch pick label

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by morrySnow** [comment]
run performance

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewRule.java [hunk 1]
```diff
@@ -801,6 +803,8 @@ protected SplitPredicate predicatesCompensate(
     private boolean containsNullRejectSlot(Set<Set<Slot>> requireNoNullableViewSlot,
             Set<Expression> queryPredicates,
             SlotMapping queryToViewMapping,
+            StructInfo queryStructInfo,
+            StructInfo viwStructInfo,
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewRule.java:807]: viwStructInfo -> viewStructInfo

**Review by xzj7019 — APPROVED** [review]

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**